src/gpu/optixPrograms.cu

```diff
@@ -163,21 +166,49 @@ extern "C" __global__ void __closesthit__()
 	}
 
 	float intensity = 0;
-	if (sbtData.textureCoords != nullptr && sbtData.texture != 0) {
-		assert(index.x() < sbtData.textureCoordsCount);
-		assert(index.y() < sbtData.textureCoordsCount);
-		assert(index.z() < sbtData.textureCoordsCount);
+	if (entityData.textureCoords != nullptr && entityData.texture != 0) {
+		assert(triangleIndices.x() < entityData.textureCoordsCount);
+		assert(triangleIndices.y() < entityData.textureCoordsCount);
+		assert(triangleIndices.z() < entityData.textureCoordsCount);
 
-		const Vec2f& uvA = sbtData.textureCoords[index.x()];
-		const Vec2f& uvB = sbtData.textureCoords[index.y()];
-		const Vec2f& uvC = sbtData.textureCoords[index.z()];
+		const Vec2f& uvA = entityData.textureCoords[triangleIndices.x()];
+		const Vec2f& uvB = entityData.textureCoords[triangleIndices.y()];
+		const Vec2f& uvC = entityData.textureCoords[triangleIndices.z()];
 
 		Vec2f uv = (1 - u - v) * uvA + u * uvB + v * uvC;
 
-		intensity = tex2D<TextureTexelFormat>(sbtData.texture, uv[0], uv[1]);
+		intensity = tex2D<TextureTexelFormat>(entityData.texture, uv[0], uv[1]);
+	}
+
+	Vec3f velocity{NAN};
+	Vec3f transformDisplacement = {0, 0, 0};
+	Vec3f skinningDisplacement = {0, 0, 0};
+	if (ctx.sceneDeltaTime > 0) {
+		if (entityData.hasPrevFrameLocalToWorld) {
+			// Computing hit point velocity in simple words:
+			// From raytracing, we get hit point in Entity's coordinate frame (hitObject).
+			// Think of it as a marker dot on the Entity.
+			// Having access to Entity's previous pose, we can compute (entityData.prevFrameLocalToWorld * hitObject),
```

Comment:
Actually it is not `FrameLocalToWorld`
This is just model (or local if U want) transform.
In that case they are the same, but only cuz parent of the entity is world origin. There is no real parenting.
When we will have parent and child in global space, then local transform of child is not equal to global transform of it, which is in that case.
Let's use proper name of frame to avoid misleading.

All the names like `localToWorld` or `someToOther`, suggest that they are not "real transform", but matrix of base change.
Those are two different concepts.
In simple scenario when You have only one transform ("position" of the object in the scene), it's somehow working, but in absolutly every other this will be a big mess, that we will not dig up from.